docs/dcp2_system_design.rst

```diff
@@ -0,0 +1,1465 @@
+the staging area.
+
+See also `Import errors`_.
+
+Invalid documents in the DCP/1
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There was one known existing schema violation in documents in the DCP/1
+production instance of DSS. The
+``provenance.schema_major_version`` and
+``provenance.schema_minor_version`` properties are present in
+metadata files submitted to the DSS from around Oct 2019 onwards. The
+addition of these fields was proposed in `RFC 11`_.
+After that RFC was accepted, the ``provenance`` schema was
+revised and the Ingest component was modified to add those fields to
+documents but the schema reference in those documents still points at an
+old schema, not the revised one. This is further complicated by the fact
+that the ``provenance`` schema is referenced indirectly via the
+main document schema (`example document`_).
+The ``analysis_file`` schema in that document is at version
+6.0.0 while the new fields were introduced in version 6.2.0 (via the
+``provenance`` schema version 1.1.0). The problem affects all
+metadata documents submitted after October of 2019, not just
+``analysis_file`` documents.
+
+.. _RFC 11: https://github.com/HumanCellAtlas/dcp-community/blob/master/rfcs/text/0011-query-by-metadata-schema-versions.md
+.. _example document: http://dss.data.humancellatlas.org/v1/files/003d3dda-6906-4943-9f12-331b963e2f55?replica=aws
+
+To address this issue, the DSS adapter removes those two fields and the
+Ingest adapter is modified to not emit them. Luckily, the fields are not
+required so removing them from documents that **do** happen to carry an
+updated schema declaration does **not** invalidate those documents. `The
+provenance schema will be revised to remove the fields
+again <provenance schema_>`_.
+
+.. _provenance schema: https://github.com/HumanCellAtlas/metadata-schema/issues/1316
+
+It was also determined that some metadata documents in DCP/1 contain
+schema URLs with a host name that points to a non-production instance of
+the site where the schemas are published, namly
+``schema.staging.data.humancellatlas.org`` and
+``schema.dev.data.humancellatlas.org``. It was established that
+the schemas referenced via these URLs were identical to their
+counterparts on ``schema.humancellatlas.org``. In order to keep
+the affected documents valid, the DCP/2 will maintain the non-production
+instances until the metadata documents are corrected and we stop
+supporting the releases affected by this issue (currently dcp1 and
+dcp2).
+
+Import errors
+-------------
+
+This work is currenly in progress and tracked by DSPDC-1604.
+
+Errors that occur during the importer's processing of the staging area
+for a particular source are logged by the importer to dedicated files in
+the staging area. The object naming scheme for these error log files is
+``errors/{timestamp}.json`` where ``timestamp`` is the
+start time of the importer invocation. The format of the file is
+`JSONL`_ but only errors should be logged. An
```

Comment:
```suggestion
`JSON Lines`_ but only errors should be logged. An
```
(nitpick of original doc) `.jsonl` is the file extension but `JSON Lines` is the format.

https://jsonlines.org/
> This page describes the JSON Lines text format, also called newline-delimited JSON